server/mdm/apple/deviceinfo.go

```diff
@@ -0,0 +1,174 @@
+// The contents of this file have been copied and modified pursuant to the following
+// license from the original source:
+// https://github.com/korylprince/dep-webview-oidc/blob/2dd846a54fed04c16dd227b8c6c31665b4d0ebd8/header/header.go
+//
+// MIT License
+//
+// Copyright (c) 2023 Kory Prince
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package apple_mdm
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha1" // nolint:gosec // See comments regarding Apple's Root CA below
+	"crypto/x509"
+	_ "embed"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/groob/plist"
+	"go.mozilla.org/pkcs7"
+)
+
+const DeviceInfoHeader = "x-apple-aspen-deviceinfo"
+
+// appleRootCert is https://www.apple.com/appleca/AppleIncRootCertificate.cer
+//
+//go:embed AppleIncRootCertificate.cer
+var appleRootCert []byte
+
+func newAppleRootCert() *x509.Certificate {
+	cert, err := x509.ParseCertificate(appleRootCert)
+	if err != nil {
+		// TODO: panic recovery?
+		panic(fmt.Errorf("could not parse cert: %w", err))
```

Comment:
I think it's fine to not recover here, this is called at startup to initialize a package var, and we control/statically embed the certificate that will be parsed, so it's something that will be raised at development time if it fails, which is what we'd want.